metarecord/tests/test_function_model.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+
+from metarecord.models import Classification, Function
+
+
+@pytest.mark.django_db
+def test_function_get_name(function):
+    assert function.get_name() == function.classification.title
+
+    function.is_template = True
+    function.save(update_fields=("is_template",))
+
+    assert function.get_name() == function.name
+
+
+@pytest.mark.django_db
+def test_function_draft_delete(function):
+    function.state = Function.APPROVED
+    can_delete = function.can_user_delete(None)
+    assert can_delete is False
+
+
+@pytest.mark.django_db
+def test_function_save_without_classification(function):
+    function.classification = None
+    try:
+        function.save()
+    except Exception as e:
+        assert str(e) == "Classification is required."
+    else:
+        assert False, "Function save should have failed!"
+
+
+@pytest.mark.django_db
+def test_approved_function_with_unapproved_classification(function):
+    function.state = Function.APPROVED
+    function.classification.state = Classification.DRAFT
+    try:
+        function.save()
+    except Exception as e:
+        assert str(e) == "Approved function must have approved classification"
+    else:
+        assert False, "Function save should have failed!"
+
+
+@pytest.mark.django_db
+def test_function_unapproved_delete_old_versions(function):
+    try:
+        function.delete_old_non_approved_versions()
+    except Exception as e:
+        assert (
+            str(e)
+            == "Function must be approved before old non-approved versions can be deleted."
+        )
+    else:
+        assert False, "Function method should have failed!"
```

Comment:
Same as https://github.com/City-of-Helsinki/helerm/pull/295/commits/e4a45642f467679cc8d25f579ffaee5fe6e2ed4c#r557091559